applications/mne_analyze/plugins/averaging/averaging.cpp

```diff
@@ -0,0 +1,633 @@
+            this, &Averaging::onChangeNumAverages, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changeBaselineFrom,
+            this, &Averaging::onChangeBaselineFrom, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changeBaselineTo,
+            this, &Averaging::onChangeBaselineTo, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changePostStim,
+            this, &Averaging::onChangePostStim, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changePreStim,
+            this, &Averaging::onChangePreStim, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changeBaselineActive,
+            this, &Averaging::onChangeBaselineActive, Qt::UniqueConnection);
+    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::resetAverage,
+            this, &Averaging::onResetAverage, Qt::UniqueConnection);
+//    connect(m_pAveragingSettingsView, &DISPLIB::AveragingSettingsView::changeStimChannel,
+//            this, &Averaging::onChangeStimChannel);
+
+    m_pAveragingSettingsView->setProcessingMode(DISPLIB::AbstractView::ProcessingMode::Offline);
+
+    // Buttons
+    QPushButton* pButton = new QPushButton();
+    pButton->setText("COMPUTE");
+    m_pAnnCheck = new QRadioButton("Average with annotations");
+    m_pStimCheck = new QRadioButton("Average with stim channels");
+
+    m_pCheckRejection = new QCheckBox("Drop Rejected");
+
+    connect(m_pCheckRejection, &QCheckBox::toggled,
+            this, &Averaging::onRejectionChecked, Qt::UniqueConnection);
+
+    connect(m_pAnnCheck, &QRadioButton::toggled,
+            this, &Averaging::onCheckBoxStateChanged, Qt::UniqueConnection);
+    connect(m_pStimCheck, &QRadioButton::toggled,
+            this, &Averaging::onCheckBoxStateChanged, Qt::UniqueConnection);
+    connect(pButton, &QPushButton::clicked,
+            this, &Averaging::onComputeButtonClicked, Qt::UniqueConnection);
+
+    //Sets Default state
+    m_pAnnCheck->click();
+    //m_pStimCheck->click();
+
+    QGroupBox* pGBox = new QGroupBox();
+    QVBoxLayout* pVBLayout = new QVBoxLayout();
+
+    pGBox->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Minimum);
+    pVBLayout->addWidget(m_pAnnCheck);
+    pVBLayout->addWidget(m_pStimCheck);
+    pVBLayout->addWidget(pButton);
+
+    pGBox->setLayout(pVBLayout);
+
+    m_pLayout->addWidget(m_pAveragingSettingsView);
+//    m_pLayout->addWidget(m_pAnnCheck);
+//    m_pLayout->addWidget(m_pStimCheck);
+    m_pLayout->addWidget(m_pCheckRejection);
+    m_pLayout->addWidget(pButton);
+
+    pWidget->setLayout(m_pLayout);
+
+    m_pTabView->addTab(pWidget, "Parameters");
+
+    pControl->setWidget(m_pTabView);
+    pControl->setAllowedAreas(Qt::LeftDockWidgetArea | Qt::RightDockWidgetArea | Qt::BottomDockWidgetArea);
```

Comment:
not sure if the vertical layout of the averaging gui would look good as a bottom dock widget? maybe only allow left and right?